### PR TITLE
Add test coverage reporting to Sonarcloud

### PR DESCRIPTION
**Pull request by danfowler**
Show overall test coverage percentage on https://sonarcloud.io/project/overview?id=omise_omise-go

See https://sonarcloud.io/summary/new_code?id=omise_omise-go&branch=submit-test-coverage

see:

* https://github.com/omise/omise-python/pull/63
* https://github.com/omise/omise-ruby/pull/54

### CODE REVIEW

File: .github/workflows/code-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    env:
+      GOPATH: ${{ github.workspace }}
+      WORKDIR: src/github.com/${{ github.repository }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Set up Go ${{ matrix.go }}
+        uses: actions/setup-go@v2
```
Comment (aashishgurung) [inline .github/workflows/code-coverage.yml:19]: Any reason why we are not using `setup-go@v3` or `setup-go@v4`?

Comment (danfowler) [reply .github/workflows/code-coverage.yml]: Resolved

File: .github/scripts/code-coverage.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  fi
+
+  echo "ok."
+}
+
+check go                       "needs go from http://golang.org"
+check $GOPATH/bin/gometalinter "needs gometalinter from https://github.com/alecthomas/gometalinter"
+
+perform linters    gometalinter --fast --aggregate
+perform tests      go test ./...  -coverprofile cover.out
```
Comment (aashishgurung) [inline .github/scripts/code-coverage.sh:43]: We are using `setup-go` action and have a `setup.sh` script to install gometalinter. Do we need this file to run linter and tests?

Comment (danfowler) [reply .github/scripts/code-coverage.sh]: No, we don't. Removed unnecessary elements of the script.

File: .github/workflows/code-coverage.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      WORKDIR: src/github.com/${{ github.repository }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go ${{ matrix.go }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.15" # Old Go version needed to run "go test"
```
Comment (danfowler) [inline .github/workflows/code-coverage.yml:21]: @aashishgurung We might need to support `go mod` to test on a more recent version of Go.

**Review by aashishgurung — APPROVED** [review]

**Review by ajzkk — APPROVED** [review]

**Pull request merged**